module/zfs/vdev.c

```diff
@@ -2385,6 +2385,23 @@ vdev_copy_path_impl(vdev_t *svd, vdev_t *dvd)
 		zfs_dbgmsg("vdev_copy_path: vdev %llu: path set to '%s'",
 		    (u_longlong_t)dvd->vdev_guid, dvd->vdev_path);
 	}
+
+	/* Our enclosure sysfs path may have changed between imports */
+	if (svd->vdev_enc_sysfs_path != dvd->vdev_enc_sysfs_path) {
+		zfs_dbgmsg("vdev_copy_path: vdev %llu: vdev_enc_sysfs_path "
+		    "changed from '%s' to '%s'", (u_longlong_t)dvd->vdev_guid,
+		    dvd->vdev_enc_sysfs_path, svd->vdev_enc_sysfs_path);
+
+		if (dvd->vdev_enc_sysfs_path)
+			spa_strfree(dvd->vdev_enc_sysfs_path);
+
+		if (svd->vdev_enc_sysfs_path) {
+			dvd->vdev_enc_sysfs_path = spa_strdup(
+			    svd->vdev_enc_sysfs_path);
+		} else {
+			dvd->vdev_enc_sysfs_path = NULL;
+		}
```

Comment:
Shouldn't we also set this in the `svd->vdev_enc_sysfs_path != NULL` case, similar to what's done for `vdev_path`?